src/selector_widget.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file selector_widget.h GUI that shows performance graphs. */
+
+#ifndef SELECTOR_WIDGET_H
+#define SELECTOR_WIDGET_H
+
+#include "stdafx.h"
+#include "querystring_gui.h"
+#include "stringfilter_type.h"
+#include "strings_type.h"
+#include "widget_type.h"
+#include "window_gui.h"
+#include "window_type.h"
+#include "safeguards.h"
+
+class SelectorWidget {
+public:
+	/**
+	 * This struct is like a configuration file for a utility,
+	 * it makes the selector widget configurable and extensible.
+	 * In it's current form it stores function pointers to functions
+	 * that rebuild the lists and draw individual items in the matrix widget
+	 */
+	struct Profile {
+		void (*DrawSection)(SelectorWidget *wid, int id, const Rect &r); ///< A function pointer to a function that draws 1 line of the list
+		void (*RebuildList)(SelectorWidget *wid); ///< A function pointer that rebuilds the item list
+	};
+
+	static const Profile company_selector_profile; ///< One prebuilt profile for selection
+	static const Profile cargo_selector_profile; ///< One prebuilt profile for selection
```

Comment:
Is this all changed for the class-based system now?

If not there's a ton of issues here...

Reply:
I did